File: Android/app/src/main/java/com/hustle/filehustle/data/TransferHistoryStore.kt
package com.hustle.filehustle.data

import androidx.compose.runtime.mutableStateListOf
import java.io.File

/**
 * Shared by the receive path (TransferServer) and send path (TransferClient)
 * so both directions land in one list. Mirrors iOS's TransferHistoryStore.
 *
 * Phase 4 TODO: persist this with Room instead of in-memory only.
 */
object TransferHistoryStore {
    val entries = mutableStateListOf<TransferHistoryEntry>()

    fun record(
        peerName: String,
        itemNames: List<String>,
        totalBytes: Long,
        direction: TransferDirection,
        status: TransferStatus,
        itemFiles: List<File> = emptyList()
    ) {
        entries.add(
            0,
            TransferHistoryEntry(
                peerName = peerName,
                itemNames = itemNames,
                totalBytes = totalBytes,
                direction = direction,
                status = status,
                itemFiles = itemFiles
            )
        )
    }
}

File: Windows/FileHustle/Data/TransferHistoryStore.cs
using System.Collections.ObjectModel;
using Avalonia.Threading;

namespace FileHustle.Data;

/// Shared by the receive path (TransferServer) and send path
/// (TransferClient) so both directions land in one list. Mirrors
/// iOS/Android's TransferHistoryStore.
///
/// Phase 4 TODO: persist this to disk instead of in-memory only.
public static class TransferHistoryStore
{
    public static ObservableCollection<TransferHistoryEntry> Entries { get; } = new();

    public static void Record(string peerName, List<string> itemNames, long totalBytes, TransferDirection direction, TransferStatus status, List<string>? itemPaths = null)
    {
        var entry = new TransferHistoryEntry
        {
            PeerName = peerName,
            ItemNames = itemNames,
            TotalBytes = totalBytes,
            Direction = direction,
            Status = status,
            ItemPaths = itemPaths ?? new List<string>(),
        };
        Dispatcher.UIThread.Post(() => Entries.Insert(0, entry));
    }
}
